### DISCUSSION TITLE
Not able to run a single Tc or specific set of Tcs from different groups

### DISCUSSION THREAD

**Discussion by SushanthN** [AppiumTestDistribution/AppiumTestDistribution#1097] [General]
@saikrishna321 In latest ATD, am able to run a set of Tcs by providing the respective Testng group name.
But do we have a support to run a specific Tc's from that group.

**Comment by saikrishna321** [comment]
@SushanthN Can you provide example?

**Reply by saikrishna321** [reply]
Does TestNG support this ?

**Reply by SushanthN** [reply]
<img width="737" alt="image" src="https://github.com/AppiumTestDistribution/AppiumTestDistribution/assets/54350763/0885d86f-6d40-40f7-918c-43c76b0efa42">

@saikrishna321  In this example, I am able to run the Sample Test class by providing group name as "demo1".
But if I want to run only Tc 'LoginTest1', I don't see a option.

**Reply by SushanthN** [reply]
@saikrishna321 Yes TesNG has the support to run specific methods from the class. Please refer the attached screenshot
<img width="865" alt="image" src="https://github.com/AppiumTestDistribution/AppiumTestDistribution/assets/54350763/a6d65488-200c-40a5-b3ee-68587fade419">


Ref link https://automationtesting.in/adding-methods-testng-xml-file/

**Comment by SushanthN22** [comment]
@saikrishna321 raised a PR for the same. I tried on my local and its working fine.
https://github.com/AppiumTestDistribution/AppiumTestDistribution/pull/1100
